SkietbaanBE/SkietbaanBE/Controllers/UserController.cs

```diff
@@ -10,50 +14,64 @@ namespace SkietbaanBE.Controllers
     public class UserController : Controller
     {
         private ModelsContext _context;
-        public UserController(ModelsContext db) {
+        public UserController(ModelsContext db)
+        {
             _context = db;
         }
-
-        //public UserController() { }
-
         // GET: api/User
         [HttpGet]
-        public IEnumerable<User> Get() {
+        public IEnumerable<User> Get()
+        {
             var users = _context.Users.ToArray<User>();
             return users;
         }
-
         // GET: api/User/5
         [HttpGet("{id}", Name = "Get")]
-        public User Get(int id) {
-            User user = _context.Users.Find(id);
+        public async Task<User> Get(int id)
+        {
+            User user = await _context.Users.FindAsync(id);
             return user;
         }
-
-        [HttpGet("test/{ret}", Name = "Test")]
-        public string Test(string ret) {
-            return $"Returned : {ret}";
-        }
-
         // POST: api/User
-        [HttpPost]
-        public virtual void Post([FromBody] User user) {
-            _context.Add(user);
-            _context.SaveChanges();
+        [HttpPost("{id}")]
+        public async Task<HttpResponseMessage> Post(int id,[FromBody] User user)
+        {
+            if (ModelState.IsValid)
+            {
+                User dbUser = await _context.Users.FindAsync(id);
+                //user not found
+                if(user == null)
```

Comment:
Was this meant to be dbUser == null?

Reply:
if the user with the specified ID cannot be found in line 41, dbUser will be null. therefore the user to be updated does not exist.